sys/shell/shell.c

```diff
@@ -343,6 +375,9 @@ int shell_handle_input_line(const shell_command_t *command_list, char *line)
             print_help(command_list);
             return 0;
         }
+        else if (IS_USED(MODULE_SHELL_JSON) && !strcmp("cmds_json", argv[0])) {
+            print_commands_json(command_list);
+        }
```

Comment:
Any reason apart from historical ones why both `help` and the new command are not defined using `SHELL_COMMAND`? This would avoid the special casing here and automatically include both in their outputs.